app/src/main/java/org/mozilla/fenix/home/intent/StartSearchIntentProcessor.kt

```diff
@@ -14,15 +14,20 @@ import org.mozilla.fenix.ext.nav
 
 /**
  * When the search widget is tapped, Fenix should open to the search fragment.
+ * Tapping the private browsing mode launcher icon should also open to the search fragment.
  */
 class StartSearchIntentProcessor(
     private val metrics: MetricController
 ) : HomeIntentProcessor {
 
     override fun process(intent: Intent, navController: NavController, out: Intent): Boolean {
-        return if (intent.extras?.getBoolean(HomeActivity.OPEN_TO_SEARCH) == true) {
-            out.putExtra(HomeActivity.OPEN_TO_SEARCH, false)
-            metrics.track(Event.SearchWidgetNewTabPressed)
+        val event = intent.extras?.getString(HomeActivity.OPEN_TO_SEARCH)
+        return if (event != null) {
+            when (event) {
+                SEARCH_WIDGET -> metrics.track(Event.SearchWidgetNewTabPressed)
```

Comment:
Is this related to the private mode PR or just adding metric to the search widget? If the latter, can you remove the changes here and update your search widget metrics PR?

Reply:
The metric for the search widget was already present. I just refactored the code a little bit here because it'll be easier to add in the telemetry for private mode PR for the future.